### ISSUE TITLE
XCode 8.0

### ISSUE THREAD

**Issue by araid** [mthiesen/Xcode_copy_line#16]
I tried updating to 8.0 by just adding the UUID, but it didn't work. Then I read that XCode 8 won't load plugins anymore: https://github.com/alcatraz/Alcatraz/issues/475

Not sure if there's any way to rewrite this a source editor extension... :/

**Comment by mthiesen** [comment]
Yeah, the Xcode 8 issue is pretty sad. I can understand Apples reasoning behind this step but the collateral damage is severe. This was bound to happen some day, plug-ins where never officially supported.

I am not sure whether one could achieve the same functionality with a source editor extension. I will look deeper into the API when I find the time.

**Comment by akabab** [comment]
hey @mthiesen, have you looked into it (editor extension) ? wanted to know before going myself

**Comment by mthiesen** [comment]
So far I didn't. Let me know what you find out.

**Comment by akabab** [comment]
Hey, I have been experimenting new xcode editor extensions and I have managed to describe the wanted behavior.. yet I need to fallback to original (built-in) copy/cut/paste calls in case there is a selection (selection.lenght > 0). I'm wondering how to trigger those original implementation, any idea?

**Comment by akabab** [comment]
here you go: https://github.com/akabab/XcodeWeshExtensions/tree/master/XcodeWeshCCP

**Comment by akabab** [comment]
So I finally managed to reproduce the original behavior and I do fallback to it if there is any selection, so now I can bind it to native keyboard shortcuts, looks like it's working. 
Developing xcode editor extensions is a pain in the ass, really.. 
Hope you can try it around and give me some feedback

**Comment by mretondo** [comment]
Does the extension work in xCode 9?
